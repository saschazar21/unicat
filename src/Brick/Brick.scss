@import '../__styles__/variables';

.wrapper {
  transition: transition(border-color);
  display: inline-flex;
  align-items: center;
  justify-content: space-between;
  padding: spacing(xs);
  border: border();
  border-color: var(--color-border-variant, transparent);
  border-radius: borderRadius();
  box-shadow: boxShadow(xs, true);
  font-family: var(--font-heading);
  font-size: fontSize(small);

  & + & {
    margin-left: spacing(xs);
  }

  svg {
    height: fontSize(h5);
    width: fontSize(h5);
    margin-right: spacing(xs);
  }
}

a.wrapper {
  --color-shadow: #{color(default, false, $systemColors, true)};
  --color-border-variant: transparent;
  box-shadow: boxShadow(s);

  &:active,
  &:hover {
    --color-border-variant: var(--color-variant);
    cursor: pointer;
  }

  &:focus {
    @include focus();
  }
}

@each $i in $variants {
  .#{$i} {
    --color-variant: #{color($i)};
    --color-border-variant: var(--color-variant);
    --color-shadow: #{color(white, false, $systemColors, true)};
    color: color($i);
    background-color: color(white, false, $systemColors);

    svg {
      fill: color($i);
    }
  }

  a.#{$i} {
    --color-variant: #{color($i, true)};
    color: color($i, true);
    background-color: color($i);

    svg {
      fill: color($i, true);
    }
  }
}
