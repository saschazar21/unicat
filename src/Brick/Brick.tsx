import React from 'react';
import classnames from 'classnames';
import { LinkIcon } from '@saschazar/unicat-icons';

import { Variant } from '../__types__/global';

import styles from './Brick.scss';

export interface BrickProps {
  /* only a string allowed as child */
  children: string;
  /* custom className */
  className?: string;
  /* icon as prefix */
  icon?: JSX.Element;
  /* hyperlink target */
  href?: string;
  /* custom variant - light by default */
  variant?: Variant;
}

export default (props: BrickProps) => {
  const {
    children,
    className: customClassName,
    icon,
    href,
    variant = 'primary',
  } = props;

  const className = classnames(
    { [styles[variant]]: variant },
    styles.wrapper,
    customClassName
  );

  return React.createElement(href ? 'a' : 'span', { className, href }, [
    href && !icon ? <LinkIcon /> : icon,
    children,
  ]);
};
