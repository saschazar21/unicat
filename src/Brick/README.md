Displays a single brick. Can be used for tags, labels, etc...

```jsx
<div style={{ display: 'flex', alignItems: 'stretch' }}>
  <Brick>I'm a brick</Brick>
  <Brick variant="default">I'm a default brick</Brick>
  <Brick href="#" variant="success">
    I'm a success brick with a link
  </Brick>
</div>
```

Of course it's possible to add an icon:

```jsx
import { StarIcon } from '@saschazar/unicat-icons';

<Brick icon={<StarIcon />}>I'm an icon brick</Brick>;
```
