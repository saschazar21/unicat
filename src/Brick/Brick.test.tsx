import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { StarIcon } from '@saschazar/unicat-icons';

import Brick from './Brick';

describe('Brick', () => {
  it('renders', () => {
    const wrapper = shallow(<Brick>I am a brick</Brick>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders an icon', () => {
    const wrapper = shallow(
      <Brick icon={<StarIcon key="demo-key" />}>I am an icon brick</Brick>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
