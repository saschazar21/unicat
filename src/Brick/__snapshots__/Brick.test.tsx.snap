// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brick renders 1`] = `
<span
  className="primary wrapper"
>
  I am a brick
</span>
`;

exports[`Brick renders an icon 1`] = `
<span
  className="primary wrapper"
>
  <StarIcon
    key="demo-key"
  />
  I am an icon brick
</span>
`;
